### PR TITLE
edit padawan

### PR DESCRIPTION
**Pull request by boykas**
add test padawan and program data to liquibase
add button edit to padawans page
add editPadawan page
add update padawan after edit

### CODE REVIEW

File: server-app/src/main/java/ua/com/findcoach/api/EditPadawanInfo.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package ua.com.findcoach.api;
+
+import ua.com.findcoach.domain.Gender;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
```
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/api/EditPadawanInfo.java]: make idea to not create such comment.

File: server-app/src/main/java/ua/com/findcoach/api/EditPadawanInfo.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * Created by DENIS on 11.01.2016.
+ */
+public class EditPadawanInfo {
```
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/api/EditPadawanInfo.java]: Is it request? Is it Response? What is it?

File: server-app/src/main/java/ua/com/findcoach/api/PadawanDTO.java [hunk 3]
```diff
@@ -1,7 +1,10 @@
 package ua.com.findcoach.api;
 
 import ua.com.findcoach.domain.Gender;
+
+import java.text.SimpleDateFormat;
 import java.util.ArrayList;
+import java.util.Date;
 import java.util.List;
 
 public class PadawanDTO {
```
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/api/PadawanDTO.java]: Use came case in class names

File: server-app/src/main/java/ua/com/findcoach/api/EditPadawanInfo.java (hunk 2 above)
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/api/EditPadawanInfo.java]: Do we change isActive status?

File: server-app/src/main/java/ua/com/findcoach/api/PadawanDTO.java (hunk 3 above)
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/api/PadawanDTO.java]: Do we change isActive status?

File: server-app/src/main/java/ua/com/findcoach/api/ProgramDTO.java [hunk 4]
```diff
@@ -14,7 +14,7 @@
         private Integer programId;
         private String programStartDate;
         private String programFinishDate;
-        private SimpleDateFormat simpleDateFormat = new SimpleDateFormat("dd-MMM-yyyy");
+        private SimpleDateFormat simpleDateFormat = new SimpleDateFormat("MM/dd/yy");
```
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/api/ProgramDTO.java]: Check Formatters class

File: server-app/src/main/java/ua/com/findcoach/controllers/CoachPadawanManagementController.java [hunk 5]
```diff
@@ -82,5 +83,41 @@ public String addPadawan(@RequestBody AddPadawanBasicInfo padawanBasicInfo) {
...
+    @RequestMapping (value = "/{padawanId}/edit-padawan.html", method = RequestMethod.GET)
+    public ModelAndView editPadawanPage(@PathVariable Integer padawanId){
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("coachAlias", coachService.retrieveCurrentCoach().getAlias());
+        Coach currentCoach = coachService.retrieveCurrentCoach();
+
+        Padawan padawan = currentCoach.getProgramList()
+        .stream()
+        .filter(program -> program.getPadawan().getPadawanId() == padawanId)
+        .findAny().get().getPadawan();
```
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/controllers/CoachPadawanManagementController.java]: What does this code do?!

File: server-app/src/main/java/ua/com/findcoach/controllers/CoachPadawanManagementController.java [hunk 6]
```diff
@@ -82,5 +83,41 @@ public String addPadawan(@RequestBody AddPadawanBasicInfo padawanBasicInfo) {
...
+
+    @RequestMapping(value = "/{padawanId}/edit-padawan.html", method = RequestMethod.POST)
+    @ResponseBody
+    public String updatePadawan(@PathVariable Integer padawanId, @RequestBody EditPadawanInfo editPadawanInfo){
+        Coach currentCoach = coachService.retrieveCurrentCoach();
+
+        Padawan padawan = currentCoach.getProgramList()
+                .stream()
+                .filter(program -> program.getPadawan().getPadawanId() == padawanId)
+                .findAny().get().getPadawan();
```
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/controllers/CoachPadawanManagementController.java]: What does this code do?

File: server-app/src/main/java/ua/com/findcoach/domain/Padawan.java [hunk 7]
```diff
@@ -29,6 +30,9 @@
     @Column
     private Gender gender;
 
+    @Column
+    private Date birthday;
+
     @ManyToOne(cascade = CascadeType.MERGE)
```
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/domain/Padawan.java]: Why is MERGE here?

File: server-app/src/main/java/ua/com/findcoach/api/EditPadawanInfo.java (hunk 1 above)
Comment (boykas) [inline server-app/src/main/java/ua/com/findcoach/api/EditPadawanInfo.java]: yes, forgot. I'll delete

File: server-app/src/main/java/ua/com/findcoach/api/EditPadawanInfo.java (hunk 2 above)
Comment (boykas) [inline server-app/src/main/java/ua/com/findcoach/api/EditPadawanInfo.java]: it's request. No we don't

File: server-app/src/main/java/ua/com/findcoach/api/PadawanDTO.java (hunk 3 above)
Comment (boykas) [inline server-app/src/main/java/ua/com/findcoach/api/PadawanDTO.java]: I didn't use camel case in this class name, becaurse I thought that DTO is abbreviation.

File: server-app/src/main/java/ua/com/findcoach/api/ProgramDTO.java (hunk 4 above)
Comment (boykas) [inline server-app/src/main/java/ua/com/findcoach/api/ProgramDTO.java]: I tried, but it didn't work

File: server-app/src/main/java/ua/com/findcoach/controllers/CoachPadawanManagementController.java (hunk 5 above)
Comment (boykas) [inline server-app/src/main/java/ua/com/findcoach/controllers/CoachPadawanManagementController.java]: This code get padawan by his Id from current coach. I made it not to connect to DB

File: server-app/src/main/java/ua/com/findcoach/controllers/CoachPadawanManagementController.java (hunk 6 above)
Comment (boykas) [inline server-app/src/main/java/ua/com/findcoach/controllers/CoachPadawanManagementController.java]: This code get padawan by his Id from current coach. I made it not to connect to DB

File: server-app/src/main/java/ua/com/findcoach/domain/Padawan.java (hunk 7 above)
Comment (boykas) [inline server-app/src/main/java/ua/com/findcoach/domain/Padawan.java]: I didn't write this string

File: server-app/src/main/java/ua/com/findcoach/api/EditPadawanInfo.java (hunk 2 above)
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/api/EditPadawanInfo.java]: Please name class correspondeny to it's function. check other classes in package.

Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/api/EditPadawanInfo.java]: Why don't we change isActive status?

File: server-app/src/main/java/ua/com/findcoach/api/PadawanDTO.java (hunk 3 above)
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/api/PadawanDTO.java]: Use camel case

File: server-app/src/main/java/ua/com/findcoach/api/ProgramDTO.java (hunk 4 above)
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/api/ProgramDTO.java]: WTF!?

File: server-app/src/main/java/ua/com/findcoach/controllers/CoachPadawanManagementController.java (hunk 5 above)
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/controllers/CoachPadawanManagementController.java]: I see what it does! But why without DB? Receive it from DB! 
ANd forget such approaches!!!

File: server-app/src/main/java/ua/com/findcoach/controllers/CoachPadawanManagementController.java (hunk 6 above)
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/controllers/CoachPadawanManagementController.java]: see previous comment

File: server-app/src/main/java/ua/com/findcoach/domain/Padawan.java (hunk 7 above)
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/domain/Padawan.java]: Ok I will check it

File: server-app/src/main/java/ua/com/findcoach/api/PadawanDtoTest.java [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+import ua.com.findcoach.domain.Gender;
+
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class PadawanDtoTest {
```
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/api/PadawanDtoTest.java]: Do we need this file?

File: server-app/src/main/webapp/WEB-INF/templates/padawan-management/padawans.html [hunk 9]
```diff
@@ -36,16 +38,28 @@
...
                 <td>$padawan.email</td>
                 <td>$padawan.gender</td>
+                <td>#if($padawan.padawanActive)
+                    Активен
+                    #else
+                    Неактивен
+                    #end
+                </td>
+                <td>
+                    <form action="/findcoach/coach/${coachAlias}/padawan-management/${padawan.padawanId}/edit-padawan.html"
```
Comment (senichas) [inline server-app/src/main/webapp/WEB-INF/templates/padawan-management/padawans.html]: Why do you use form here? 
Use simple link!

File: server-app/src/main/webapp/WEB-INF/templates/padawan-management/edit-padawan.html [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+                <p><input type="radio" name="padawanGender" value="FEMALE" id="padawanGenderFemale"
+                          ng-model="padawanData.gender"/>&nbsp;<label for="padawanGenderFemale">Женский</label></p>
+
+                <p>Год рождения</p>
+
+                <p><input type="date" name="padawanYear" id="padawanYear" ng-model="padawanData.birthday"/></p>
+
+                <p>Активность</p>
+
+                <p><input type="radio" name="padawanActive" value="true" id="padawanActiveTrue"
```
Comment (senichas) [inline server-app/src/main/webapp/WEB-INF/templates/padawan-management/edit-padawan.html]: For active status use check box

File: server-app/src/main/webapp/WEB-INF/templates/padawan-management/edit-padawan.html [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+</head>
+<body>
+<div id="mainContainer">
+    <header id="coachHeader">
+        <h1><a href="/findcoach/coach/profile/dashboard.html">Редактирование падавана</a></h1>
+    </header>
+    #parse("padawan-management/menu.vm")
+    <section>
+        <form ng-controller="editPadawanController" ng-submit="submitEditPadawan()" ng-init="padawanData.firstName= '$padawan.firstName';
+        padawanData.lastName = '$padawan.lastName'; padawanData.email = '$padawan.email'; padawanData.gender = '$padawan.gender';
```
Comment (senichas) [inline server-app/src/main/webapp/WEB-INF/templates/padawan-management/edit-padawan.html]: extract ng-init as separate js method.

File: server-app/src/main/java/ua/com/findcoach/api/PadawanDtoTest.java (hunk 8 above)
Comment (boykas) [inline server-app/src/main/java/ua/com/findcoach/api/PadawanDtoTest.java]: No. I have already deleted it

File: server-app/src/main/webapp/WEB-INF/templates/padawan-management/padawans.html (hunk 9 above)
Comment (boykas) [inline server-app/src/main/webapp/WEB-INF/templates/padawan-management/padawans.html]: I wanted to use button. Ok, I'll make without it

File: server-app/src/main/java/ua/com/findcoach/controllers/CoachPadawanManagementController.java [hunk 12]
```diff
@@ -82,5 +82,36 @@ public String addPadawan(@RequestBody AddPadawanBasicInfo padawanBasicInfo) {
         return "";
     }
 
+    @RequestMapping(value = "/{padawanId}/edit-padawan.html", method = RequestMethod.GET)
+    public ModelAndView editPadawanPage(@PathVariable Integer padawanId) {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("coachAlias", coachService.retrieveCurrentCoach().getAlias());
+        Padawan padawan = padawanService.findById(padawanId);
+
+        parameters.put("padawan", new PadawanDto(padawan.getPadawanId(),
```
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/controllers/CoachPadawanManagementController.java]: I've just noticed this is incorrect usage of DTO. In this case Padawan can be used.

File: server-app/src/main/java/ua/com/findcoach/controllers/CoachProgramController.java [hunk 13]
```diff
@@ -50,24 +50,27 @@ public ModelAndView recieveCoachProgramPadawans(@PathVariable String coachAlias)
                 .entrySet().stream()
                 .forEach(entry ->
                 {
-                    PadawanDTO padawanDTO = new PadawanDTO(
+                    PadawanDto padawanDto = new PadawanDto(
```
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/controllers/CoachProgramController.java]: Here is also.
Just Padawan POJO can be used here.

Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/controllers/CoachProgramController.java]: Here is also. PAdawan POJO can be used.

File: server-app/src/main/webapp/WEB-INF/templates/padawan-management/edit-padawan.html [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+    <title>Редактирование падавана</title>
+</head>
+<body>
+<div id="mainContainer">
+    <header id="coachHeader">
+        <h1><a href="/findcoach/coach/profile/dashboard.html">Редактирование падавана</a></h1>
+    </header>
+    #parse("padawan-management/menu.vm")
+    <section>
+        <form ng-controller="editPadawanController" ng-submit="submitEditPadawan()" ng-init="test = $padawan.birthday">
```
Comment (senichas) [inline server-app/src/main/webapp/WEB-INF/templates/padawan-management/edit-padawan.html]: Is it correct information in method ng-init?

File: server-app/src/main/java/ua/com/findcoach/api/ProgramDto.java [hunk 15]
```diff
@@ -0,0 +1,66 @@
+package ua.com.findcoach.api;
+
+import ua.com.findcoach.domain.Goal;
+
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Date;
+
+/**
```
Comment (senichas) [inline server-app/src/main/java/ua/com/findcoach/api/ProgramDto.java]: Comments!

**Pull request merged**